handler/src/main/java/io/netty/handler/ssl/TlsMasterKeyHandler.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+import javax.crypto.SecretKey;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLSession;
+import java.lang.reflect.Field;
+
+/**
+ * The {@link TlsMasterKeyHandler} is a channel-handler you can include in your pipeline to consume the master key
+ * & session identifier for a TLS session.
+ * This can be very useful when combined with {@link TlsMasterKeyBiConsumer.WiresharkTlsMasterKeyBiConsumer} as it will
+ * log the secret & identifier in a format that is consumable by Wireshark -- allowing easy decryption of pcap/tcpdumps.
+ */
+public class TlsMasterKeyHandler extends ChannelInboundHandlerAdapter {
+
+    /**
+     * The JRE SSLSessionImp cannot be imported
+     */
+    private static final Class<?> SSL_SESSIONIMPL_CLASS;
+
+    /**
+     * A system property that can be used to turn on/off the {@link TlsMasterKeyHandler} dynamically without having
+     * to edit your pipeline.
+     * <code>-Dio.netty.tls.masterKeyHandler=true</code>
+     */
+    private static final String SYSTEM_PROP_KEY = "io.netty.tls.masterKeyHandler";
+
+    static {
+        try {
+            SSL_SESSIONIMPL_CLASS = Class.forName("sun.security.ssl.SSLSessionImpl");
```

Comment:
imho we should do something similar to what we do in `OpenSsl.java` in terms of init.

Reply:
I believe I've done something similar now according to your feedback.